.devcontainer/devcontainer.json

```diff
@@ -7,7 +7,12 @@
     "ghcr.io/eitsupi/devcontainer-features/duckdb-cli:0": {}
   },
   // Use 'postCreateCommand' to run commands after the container is created.
-  "postCreateCommand": "python -m pip install --user -r requirements.txt && dbt deps",
+  "postCreateCommand": "python -m pip install --user -r requirements.txt && dbt deps && npm --prefix ./reports install",
+  // Set environment variables for evidence database connection
+  "remoteEnv": {
+    "DATABASE": "duckdb",
+    "FILENAME": "jaffle_shop.duckdb"
+  },
```

Comment:
Does this hard code the values permanently or can they still be passed from the parent caller's context?

One of the nice things about calling via `meltano invoke evidence:dev` is that the Meltano [evidence-ext](https://hub.meltano.com/utilities/evidence/) plugin automatically passes all the credentials and config info to the Evidence context. It does this by creating an evidence config file here: https://github.com/MeltanoLabs/evidence-ext/blob/d53b44206bb83cf98adb2b81c1fc6c7e3f1f6793/evidence_ext/config.py#LL53

This invocation pattern is the draft proposal:

- #9

Calling with Meltano gets around having to reconfigure upon each launch, while still being dynamic and able to support a variety of configuration patterns without needing the "settings" screen on the web UI.

In terms of having sensible defaults for non-Meltano invocations, is there a way to keep your and additions and and still override them at runtime?

Cc @kgpayne

Reply:
Also worth noting that `evidence-ext` also calls [npm install](https://github.com/MeltanoLabs/evidence-ext/blob/d53b44206bb83cf98adb2b81c1fc6c7e3f1f6793/evidence_ext/extension.py#L126) before `build` or `dev`. This may be configurable in the future (to minimise the penalty for automated build actions in stable/frozen environments) but for now means that the developer experience with Meltano is super streamlined. Folks unfamiliar with npm need not be exposed to it at all 😅

Reply:
Just played around in the meltano branch - that is a very streamlined experience!

I did some testing on these hardcoded env vars: they are set when the codespace is built, but can be overridden afterwards. I tested setting the env vars and then running the meltano commands, and evidence picks up the correct variables from meltano in that case.

Reply:
> Just played around in the meltano branch - that is a very streamlined experience!

That's great. Glad to hear it! I think this project is a great example of where meltano can add value by streamlining config and connections between the tools. (A like but like Make, I guess, but for data projects 😁.)

> I did some testing on these hardcoded env vars: they are set when the codespace is built, but can be overridden afterwards. I tested setting the env vars and then running the meltano commands, and evidence picks up the correct variables from meltano in that case.

Fantastic. Thanks very much for checking that. That was my only concern.

Reply:
@gwenwindflower - This looks good to me. Per the above, I think the comment from @hughess addresses the only concern I had.